### ISSUE TITLE
GetText.NET.Extractor:  not all String.Format C# are marked as csharp-format

### ISSUE THREAD

**Issue by mcka-dev** [perpetualKid/GetText.NET#16]
Hi
in GetText.NET.Extractor v0.9.6.0 not all [String.Format](https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netframework-4.8) C# are marked as `#, csharp-format`

**Source:**
```
using System;
using System.Windows.Forms;

namespace HelloApp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            labelTotalSize.Text = totalDiscSize < 1000000000 ?
                catalog.GetString("{0} MB", totalDiscSize / 1000000) :
                catalog.GetString("{0:F2} GB", (float)totalDiscSize / 1000000000.0);
        }
    }
}
```

**Actual result:**
```
msgid ""
msgstr ""
"Project-Id-Version: 1\n"
"POT-Creation-Date: 2020-11-12 10:52:56+0300\n"
"PO-Revision-Date: 2020-11-12 10:52:56+0300\n"
"Last-Translator: \n"
"Language-Team: \n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"X-Generator: GetText.NET Extractor\n"

#: .\1\1.cs:16
#: .\1\1.cs:16
msgid "{0} MB"
msgstr ""

#: .\1\1.cs:17
#: .\1\1.cs:17
msgid "{0:F2} GB"
msgstr ""
```
and we also got a duplicate link.

**Expected Result:**
```
msgid ""
msgstr ""
"Project-Id-Version: 1\n"
"POT-Creation-Date: 2020-11-12 10:52:56+0300\n"
"PO-Revision-Date: 2020-11-12 10:52:56+0300\n"
"Last-Translator: \n"
"Language-Team: \n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"X-Generator: GetText.NET Extractor\n"

#: .\1\1.cs:16
#, csharp-format
msgid "{0} MB"
msgstr ""

#: .\1\1.cs:17
#, csharp-format
msgid "{0:F2} GB"
msgstr ""
```

**Comment by perpetualKid** [comment]
given that's in the comments sections only, this shouldn't cause any functional issues.

It's indeed missing the `#, csharp-format` marker for traditional C# format strings (`"{0}, param"`) but works correct for interpolated strings (`$"{param}"`)

As for the duplicate reference, this is kinda correct in the logic of the extractor - actually it does find the strings twice, once as it scans the Windows.Forms.Controls Text property (`labelTotalSize.Text = ""`), and the other time through the `catalog.GetString("")`  calls, I'll add some check for that.

But given thats more cosmetical changes, I'm hesitant to release a new package just for that fix, unless this is blocking you,

**Comment by perpetualKid** [comment]
solved in 0.9.7-beta.1
